client/core/account.go

```diff
@@ -7,6 +7,39 @@ import (
 	"github.com/decred/dcrd/dcrec/secp256k1/v3"
 )
 
+// AccountDisable is used to disable an account by given host and application
+// password.
+func (c *Core) AccountDisable(pw []byte, host string) error {
+	_, err := c.encryptionKey(pw)
+	if err != nil {
+		return codedError(passwordErr, err)
+	}
+	_, err = addrHost(host)
+	if err != nil {
+		return newError(addressParseErr, "error parsing address: %v", err)
+	}
+
+	// Get the dexConnection.
+	c.connMtx.RLock()
+	dc, found := c.conns[host]
+	if !found {
+		return newError(unknownDEXErr, "DEX: %s", host)
+	}
+
+	acctInfo, err := c.db.Account(dc.acct.host)
+	if err != nil {
+		return newError(accountRetrieveErr, "Error retrieving account: %v", err)
+	}
+	err = c.db.DisableAccount(acctInfo)
+	if err != nil {
+		return newError(accountDisableErr, "Error disabling account: %v", err)
+	}
+	c.conns[host].connMaster.Disconnect()
+	delete(c.conns, host)
```

Comment:
I know this is in draft, but a few observations: (1) you never unlock the connMtx, (2) you have `dc`, so no need to do `c.conns[host].connMaster.Disconnect()`, and (3) I don't think we want to do any of this while the dexConnection has active trades, so we should probably check first.